GoogleForCreators/web-stories-wp
# Auto-hide scrollbars

## Summary

Respects the OS-level auto-hide setting for scrollbars.

Tested on FF, Chrome and Safari with and without obtrusive scrollbar setting.

Compare with the following CodePens:

https://codepen.io/zachleat/pen/GRKxzqB (default behavior)
https://codepen.io/swissspidy/pen/vYNvOVd (same logic as in this PR)

## Relevant Technical Choices

Adds a new hook to detect scrollbar behavior as set by the OS, as originally suggested on #1698.

## To-do


## User-facing changes

Scrollbars set to auto-hide: the scrollbars are only visible when scrolling, but are correctly styled as before.

Scrollbars set to always show: no changes. The scrollbars are always shown, styled as before.

## Testing Instructions
With Mac:
I
1. In System Preferences, choose "Automatically based on mouse or trackpad" as the option for "Show scroll bars"
2. Verify that by default the scrollbars are not visible (refresh the editor if you had it open before)
3. Verify that the scrollbars appear when scrolling, test this in Media Library, Design Panel, Color Presets (need to have a few lines of presets)
4. Verify the same for Chrome, Firefox and Safari

II
1. In System Preferences, choose "Always" as the option for "Show scroll bars"
2. Verify that all the scrollbars are visible by default ((refresh the editor if you had it open before)
3. Verify the same for Chrome, Firefox and Safari

---



Fixes #1698
